server/src/main/java/org/elasticsearch/index/stats/IndexingPressureStats.java

```diff
@@ -0,0 +1,80 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.stats;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentFragment;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+public class IndexingPressureStats implements Writeable, ToXContentFragment {
+
+    private final long totalPrimaryAndCoordinatingBytes;
+    private final long totalReplicaBytes;
+    private final long pendingPrimaryAndCoordinatingBytes;
+    private final long pendingReplicaBytes;
+    private final long primaryAndCoordinatingRejections;
+    private final long replicaRejections;
+
+    public IndexingPressureStats(StreamInput in) throws IOException {
+        totalPrimaryAndCoordinatingBytes = in.readVLong();
+        totalReplicaBytes = in.readVLong();
+        pendingPrimaryAndCoordinatingBytes = in.readVLong();
+        pendingReplicaBytes = in.readVLong();
+        primaryAndCoordinatingRejections = in.readVLong();
+        replicaRejections = in.readVLong();
+    }
+
+    public IndexingPressureStats(long totalPrimaryAndCoordinatingBytes, long totalReplicaBytes, long pendingPrimaryAndCoordinatingBytes,
+                                 long pendingReplicaBytes, long primaryAndCoordinatingRejections, long replicaRejections) {
+        this.totalPrimaryAndCoordinatingBytes = totalPrimaryAndCoordinatingBytes;
+        this.totalReplicaBytes = totalReplicaBytes;
+        this.pendingPrimaryAndCoordinatingBytes = pendingPrimaryAndCoordinatingBytes;
+        this.pendingReplicaBytes = pendingReplicaBytes;
+        this.primaryAndCoordinatingRejections = primaryAndCoordinatingRejections;
+        this.replicaRejections = replicaRejections;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        // TODO: Add total
+        out.writeVLong(totalPrimaryAndCoordinatingBytes);
+        out.writeVLong(totalReplicaBytes);
+        out.writeVLong(pendingPrimaryAndCoordinatingBytes);
+        out.writeVLong(pendingReplicaBytes);
+        out.writeVLong(primaryAndCoordinatingRejections);
+        out.writeVLong(replicaRejections);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject("indexing_pressure");
+        builder.field("total_primary_and_coordinating_bytes", totalPrimaryAndCoordinatingBytes);
+        builder.field("total_replica_bytes", totalReplicaBytes);
```

Comment:
should we have another `total_bytes` field here that sums `total_primary_and_coordinating_bytes` and `total_replica_bytes`?

Reply:
I can add that if you want. However, then do we need a `total_pending_bytes`? I guess I did not add it in the first place as the value you are looking to add is computed by simply adding the two numbers and it felt unnecessary to add totals.

Reply:
It's mainly for convenience, so you can get a quick idea of the indexing pressure on all nodes without extra aggregating (possibly the user might not care what exact bucket (primary, coordination, replica) the memory is for.

We would also have `total_pending_bytes`, yes (that was actually the one I was mainly looking for).

Further I prefer the `current_` prefix to `pending_`, we use `current_` in a number of related stats places as well.

Reply:
Changed to current